### PR TITLE
First compilation of krel documentation

### LINKED ISSUE #846: build scripts README.md
The https://github.com/kubernetes/release/tree/master/build subdirectory of scripts needs a README.md describing the tools, example invocations, expected outputs, tests.

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:
This is the firs batch of the krel documentation. Most of the subcommand docs are derived from krels help output but the idea is to have a first version of the docs from where we can improve them.

ref: #846

#### Special notes for your reviewer:
This is the first version of the docs. As noted above, I wrote this first batch of documents following the --help output of each subcommand. This bunch of file provides the structure to build a full featured set of documents.

If you or someone you know can provide insight to add more details, caveats, known issues or other important information please let me know.

#### Does this PR introduce a user-facing change?

```release-note
First batch of krel usage documents. Main README.md and a doc for each subcommand.
```

### CODE REVIEW

File: docs/krel/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+| [changelog](changelog.md) | Automate the lifecycle of CHANGELOG-x.y.{md,html} files in a k/k repository |
+| [ff](ff.md) | Fast forward a Kubernetes release branch |
+| [gcbmgr](gcbmgr.md) | Submit Kubernetes staging and release jobs to Google Cloud Build |
+| [patch-announce](patch-announce.md) | Send out patch release announcement mails |
+| [push](push.md) | Push Kubernetes release artifacts to Google Cloud Storage (GCS) |
+| [release-notes](release-notes.md) | The subcommand of choice for the Release Notes subteam of SIG Release |
+
+## Important Notes
+
+**Note:** Some of the krel subcommands are under development and their usage may already differ from these docs.
```
Comment (saschagrunert) [inline docs/krel/README.md]:
```suggestion
Some of the krel subcommands are under development and their usage may already differ from these docs. 
```

File: docs/krel/changelog.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      --bucket string      Specify gs bucket to point to in generated notes (default "kubernetes-release")
+  -h, --help               help for changelog
+      --html-file string   The target html file to be written. If empty, then it will be CHANGELOG-x.y.html in the current path.
+      --record string      Record the API into a directory
+      --replay string      Replay a previously recorded API from a directory
+      --tag string         The version tag of the release, for example v1.17.0-rc.1
+      --tars string        Directory of tars to SHA512 sum for display (default ".")
+```
+
+## Important Notes
```
Comment (saschagrunert) [inline docs/krel/changelog.md]:
```suggestion
```

File: docs/krel/patch-announce.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Currently, `patch-announce` will notify the following addresses:
+| Recipient | E-Mail Address |
+| --------- | -------------- |
+| Kubernetes developer/contributor discussion | kubernetes-dev@googlegroups.com |
+| kubernetes-dev-announce | kubernetes-dev-announce@googlegroups.com |
+
+
+## Important Notes
+
```
Comment (saschagrunert) [inline docs/krel/patch-announce.md]:
```suggestion
```

**Comment by puerco** [comment]
Ready !

**Comment by justaugustus** [comment]
Great work, @puerco!
/lgtm
/approve

**Pull request merged**